packages/community/groq/src/provider.ts

```diff
@@ -0,0 +1,312 @@
+import type { Logger } from '@nestjs/common';
+import type {
+  CompletionMetadata,
+  ICompletionProvider,
+  CompletionRequest,
+  CompletionResponse,
+  ResourceModelConfig,
+  AIConnection,
+  TokenMessage,
+  AIProviderConfig,
+} from '@vm-x-ai/completion-provider';
+import { BaseCompletionProvider, TokenCounter } from '@vm-x-ai/completion-provider';
+import Groq from 'groq-sdk';
+import { Stream } from 'groq-sdk/lib/streaming';
+import type {
+  ChatCompletionCreateParams,
+  ChatCompletionCreateParamsNonStreaming,
+  ChatCompletionCreateParamsStreaming,
+  ChatCompletionUserMessageParam,
+  ChatCompletionAssistantMessageParam,
+  ChatCompletionSystemMessageParam,
+} from 'groq-sdk/resources/chat/completions';
+import { Span } from 'nestjs-otel';
+import type { Subject } from 'rxjs';
+
+const API_TIMEOUT_SECONDS = 300;
+
+export type CallAPIRequest = ChatCompletionCreateParamsNonStreaming | ChatCompletionCreateParamsStreaming;
+
+export type GroqConnectionConfig = {
+  apiKey: string;
+};
+
+export class GroqLLMProvider extends BaseCompletionProvider<Groq> implements ICompletionProvider {
+  constructor(logger: Logger, provider: AIProviderConfig) {
+    super(logger, provider);
+  }
+
+  getMaxReplyTokens(request: CompletionRequest): number {
+    const maxTok = request.config?.max_tokens || 100; // REPLACE THIS!!
```

Comment:
Should we replace the `100` value with something else?

Reply:
let's discuss this